astropy/io/fits/hdu/table.py

```diff
@@ -1294,7 +1294,7 @@ def _dump_coldefs(self, fileobj):
         for column in self.columns:
             line = [column.name, column.format]
             attrs = ['disp', 'unit', 'dim', 'null', 'bscale', 'bzero']
-            line += ['{:16s}'.format(value if value else '""')
+            line += [f'{str(value):16s}' if value else '""'
```

Comment:
Do you mind if I keep the 'f-string' formatting... If I understand correctly no python older than 3.6 is supported by astropy... right?

Reply:
The problem is that the empty string case should also be padded, and I think would be clearer with `.format` in this case (you could put the if/else in the fstring but that would make multiple levels of quote which is not ideal).